src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -235,7 +236,7 @@ CleanupTaskExecution(TaskExecution *taskExecution)
 	pfree(taskExecution->taskStatusArray);
 	pfree(taskExecution->connectionIdArray);
 	pfree(taskExecution->fileDescriptorArray);
-	memset(taskExecution, 0, sizeof(TaskExecution));
+	pfree(taskExecution);
```

Comment:
Note this is not equivalent - have you checked the callers that there's no references to the task execution itself after this?